src/components/Rubric.js

```diff
@@ -59,52 +74,18 @@ class Rubric extends Component {
     const index = this.state.activeCompetencyIndex;
     const scales = [];
     if (this.props.rubric.Competencies) {
-      const scales = this.props.rubric.Competencies[index];
-    } 
-
-    const levelNames = ['Initial', 'Approaching', 'Overtaking', 'Innovating'];
-    // iterate through the four levels
-    // and add header for each level
-    // iterate through the scales for the current Competency
-    // and add the criteria that have the level that matches the level we're on
-      // get the criteria for the given level
-      var criteria = [];
-      for (var j = 0; j < scales.length; j++) {
-        console.log(scales[j])
-        // each element in scales is a criteria object
-        // so get the criteria for which the level is each to the level we're on
-        for (var i = 0; i < scales[j].Criteria.length; i++) {
-          console.log(scales[j].Criteria[i].level)
-          if (scales[j].Criteria[i]= level) {
-            criteria.push(
-              <tr>
-                <td>
-                  <div className="checkbox">
-                    <label>
-                      <input type="checkbox" name="optionsCheckboxes" defaultChecked="" />
-                    </label>
-                  </div>
-                </td>
-                <td>{scales[j].name}</td>
-              </tr>
-            )
-          }
-        }
-        }
+      return this.props.rubric.Competencies[index].Scales.map((scale, index) => {
+        // if the criteria level matches the level parameter, add the
+        // criteria component
+        return scale.Criteria.filter(criteria => criteria.level == level).map((criteria, index) => {
```

Comment:
in the getCompetencyButton, I like it that we correctly applied .filter .map when we return the filter result for criteria component.